TFT/src/User/API/Vfs/vfs.c

```diff
@@ -132,63 +132,88 @@ bool IsRootDir(void)
 }
 
 // check if filename provides a supported filename extension
-char * isSupportedFile(char * filename)
+char *isSupportedFile(char * filename)
 {
-  char * extPos = strstr(filename, ".g");  // support "*.g","*.gco" and "*.gcode"
+  char *extPos = strrchr(filename, '.');  // check last "." in the name where extension is supposed to start
 
-  if (extPos == NULL)
-    extPos = strstr(filename, ".G");  // support "*.g","*.gco" and "*.gcode"
+  if (extPos != NULL && extPos[1] != 'g' && extPos[1] != 'G')
+  {
+    extPos = NULL;
+  }
 
   return extPos;
 }
-
-char * hideFileExtension(uint8_t index)
+/**
+  * @brief  Temporarily remove extension 
+  * @param  filename: targeted filename
+  * @param  maxLength: the max length of the "filename" parameter as it was defined.
+  *         It is needed to check if extension hide flag can be added.
+  * @retval pointer to the resulting name
+  */
+char *hideFileExtension(char * filename)
 {
-  char * filename = infoFile.file[index];
   char * extPos;
 
-  if (infoSettings.filename_extension == 0)  // if filename extension is disabled
+  extPos = isSupportedFile(filename);
+  if (extPos != NULL)  // if filename provides a supported filename extension
+  {
+    filename[extPos - filename] = 0;  // temporarily hide filename extension
+  }
+  else
```

Comment:
as reported before, no need (and it could cause a bug) to set extension flag here. The "else" statement must be removed